typescript/web/src/components/toast.ts

```diff
@@ -1,43 +1,98 @@
 import { ApolloError } from "@apollo/client";
-import { AlertStatus, useToast as useChakraToast } from "@chakra-ui/react";
+import {
+  AlertStatus,
+  useToast as useChakraToast,
+  UseToastOptions as UseChakraToastOptions,
+} from "@chakra-ui/react";
+import { isNil } from "lodash/fp";
 import { useCallback } from "react";
 import { getApolloErrorMessage } from "../utils/get-apollo-error-message";
 
+export type ToastFnParameters =
+  | [UseChakraToastOptions]
+  | [string, string, AlertStatus]
+  | [];
+
+export type StatusToastFnParameters =
+  | [Omit<UseChakraToastOptions, "status">]
+  | [string, string]
+  | [];
+
+const getToastOptions = (
+  [titleOrOptions, ...args]: ToastFnParameters | StatusToastFnParameters,
+  hookOptions: UseChakraToastOptions | undefined = {}
+): UseChakraToastOptions => {
+  if (isNil(titleOrOptions)) return hookOptions;
+  if (typeof titleOrOptions === "object") {
+    return { ...hookOptions, ...titleOrOptions };
+  }
+  const [description = hookOptions?.description, status = hookOptions?.status] =
+    args;
+  return { ...hookOptions, title: titleOrOptions, description, status };
+};
+
 /**
  * Returns a toast reusable across the application
  */
-export const useToast = () => {
-  const toast = useChakraToast();
+export const useToast = (...hookArgs: ToastFnParameters) => {
+  const toastOptions = getToastOptions(hookArgs);
+  const toast = useChakraToast({
+    // Make description actually optional since when title is **not** empty:
+    // * Title is hidden if description is `undefined`
+    // * Title is shown if description is an empty string
+    description: "",
+    ...toastOptions,
+  });
   return useCallback(
-    (title: string, description: string, status: AlertStatus) =>
-      toast({
-        title,
-        description,
-        status,
+    (...fnArgs: ToastFnParameters) => {
+      const fnOptions = getToastOptions(fnArgs);
+      return toast({
         isClosable: true,
         position: "bottom-right",
         duration: 10000,
```

Comment:
Yes I agree but we don't have a notification center yet so it's the only way to make sure that the user had enough-time to notice the toast and eventually hover it so that it doesn't disappear.